### PR TITLE
fix RangeBitmap#between bug in full section after empty section

### PR DESCRIPTION
**Pull request by richardstartin**
### SUMMARY
- fixes a bug in between queries in full sections immediately after empty sections. This would be triggered when the `Bits` class's bitset gets out of sync with its empty and full flags.
- Adds a regression test reduced from data collected from Apache Pinot
- Re-enables some tests which I verified to be very fast but were disabled along with the very heavy tests (I saw the message at the top of `RangeBitmapTest` - mea culpa)

Many thanks to @klsince for giving me a data set I could reproduce this (the smaller regression case is reduced from this).

### Automated Checks

- [x] I have run `./gradlew test` and made sure that my PR does not break any unit test.
- [x] I have run `./gradlew checkstyleMain` or the equivalent and corrected the formatting warnings reported.

### CODE REVIEW

File: RoaringBitmap/src/main/java/org/roaringbitmap/RangeBitmap.java [hunk 1]
```diff
@@ -1308,20 +1306,18 @@ public void reset(int boundary) {
...
         resetBitmapRange(bits, boundary, 0x10000);
       }
-      empty = false;
-      full = false;
+      makeNonEmpty();
+      makeNonFull();
     }
 
     public void flip(int from, int to) {
+      Util.flipBitmapRange(bits, from, to);
```
Comment (richardstartin) [inline RoaringBitmap/src/main/java/org/roaringbitmap/RangeBitmap.java:1314]: This is the fix

**Comment by richardstartin** [comment]
@lemire are you ok with me re-enabling some of the tests you disabled? I accept some of them were far too heavy, but I ran them all and the ones I re-enabled all took less than 20s cpu time, and most took less than5s cpu time, all less than 2s wall time on a laptop.

**Comment by lemire** [comment]
> are you ok with me re-enabling some of the tests you disabled?

Sure. I think it is a balancing act between having tests that are so slow that you are causing climate change and letting bugs through.

Ideally, the tests should catch all bugs, but that's impossible... so we have to hit the right level, knowing that we eventually have diminishing returns...

**Comment by lemire** [comment]
Looks credible. Merging.

**Pull request merged**